### ISSUE TITLE
Add a way to create the connectors in cp-kafka-connect while depoying

### ISSUE THREAD

**Issue by amit-k-yadav** [confluentinc/cp-helm-charts#390]
## Description
* As far as I know, in the distributed mode of Kafka connect (which this Helm chart supports) the only way to create connectors is to send a request to the connect endpoint (e.g http://localhost:8083/connectors).
* When deploying Kafka connect on kubernetes, to create the connectors we either:
    1. Need to use [kubectl exec](https://kubernetes.io/docs/tasks/debug-application-cluster/get-shell-running-container/) to go inside the container and then make the requests to the endpoint. In short, use the REST API from inside the pod.
    2. Or have an external service to reach this endpoint for us (a different pod deployed by us or manually reaching the Kafka-connect pod IP). In short, use the REST API from outside the pod.
    3. Use confluent CLI (which is only recommended in development mode)
* In short, this step is not automated with the Helm chart (cp-kafka-connect)

## Suggestion
* To support running custom scripts in the `command` argument of the container so that an external script can contain the necessary code to create the connector after the endpoint gets ready.

## Example
* As this issue ([467](https://github.com/confluentinc/cp-docker-images/issues/467)) on [confluentinc/cp-docker-images](https://github.com/confluentinc/cp-docker-images) mentions we can add commands after triggering the ENTRYPOINT of the container.
* There are plenty of alternative solutions given in the same issue for docker-compose, something similar in cp-kafka-connect chart would be really helpful.

**Comment by shcallaway** [comment]
Sure, it would be nice if the chart created your connectors upon installation, but remember that connectors are persistent, so you only have to create them once. It could cause problems if your chart is created new (duplicate) connectors every time it is installed.

**Comment by amit-k-yadav** [comment]
> Sure, it would be nice if the chart created your connectors upon installation, but remember that connectors are persistent, so you only have to create them once. It could cause problems if your chart is created new (duplicate) connectors every time it is installed.

It makes perfect sense. This is the reason why the external scripts should be allowed to be mounted and run. We can add a logic in the exernal script to check if the connector exists and create only when it doesn't exist.

I have created a PR (here is the [link](https://github.com/confluentinc/cp-helm-charts/pull/392)) which solves this issue.

* Example script:

```shell
echo "Waiting for Kafka Connect to start listening on kafka-connect  "
while :; do
    # Check if the connector endpoint is ready
    # If not check again
    curl_status=$(curl -s -o /dev/null -w %{http_code} http://localhost:{{ .Values.servicePort }}/connectors)
    echo -e $(date) "Kafka Connect listener HTTP state: " $curl_status " (waiting for 200)"
    if [ $curl_status -eq 200 ]; then
        break
    fi
    sleep 5
done

echo "======> Creating connectors"
# Send a simple POST request to create the connector
curl -X POST \
    -H "Content-Type: application/json" \
    --data '{
    "name": "sample-connector",
    "config": {
        "connector.class": "io.confluent.connect.jdbc.JdbcSourceConnector",
        "tasks.max": 1,
        "connection.url": "jdbc:mysql://123.4.5.67:3306/test_db?user=root&password=pass",
        "mode": "incrementing",
        "incrementing.column.name": "id",
        "timestamp.column.name": "modified",
        "topic.prefix": "sample-connector-",
        "poll.interval.ms": 1000
        }
    }' http://$CONNECT_REST_ADVERTISED_HOST_NAME:8083/connectors
```

If you look at the above script it doesn't really check for the existence of the connector but if it does already the script will get a `409 Conflict` status code instead of `202 Success`. This won't make the container restart because in this because the PR also supports liveness probe and which looks for the Kafka-connect port liveness.
